src/main/java/seedu/address/logic/commands/SummariseCommand.java

```diff
@@ -0,0 +1,81 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Faculty;
+import seedu.address.model.person.Person;
+
+/**
+ * Summarise all the students by faculty and show how many students in that faculty are positive, negative, HRW, HRN
+ */
+public class SummariseCommand extends Command {
+
+    public static final String COMMAND_WORD = "summarise";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Summarise all data in the address book and presents an overview of the covid situation in school.\n";
+
+    public static final String MESSAGE_SUMMARISE_PERSON_SUCCESS = "Summarising all students: \n";
+
+    private static final String FACULTY_SUMMARY_FORM = "\nIn %s with %d student(s),\n"
+            + "Covid Positive: %d student(s)\n"
+            + "Covid Negative: %d student(s)\n"
+            + "Health Risk Warning: %d student(s)\n"
+            + "Health Risk Notice: %d student(s)\n"
+            + "%.2f percent of student(s) here are suffering...\n";
+
+    private static final List<String> FACULTIES = Faculty.getFacultyEnumAsList();
+
+    private static final Predicate<Person> BY_POSITIVE = person -> person.getStatusAsString().equals("POSITIVE");
+    private static final Predicate<Person> BY_NEGATIVE = person -> person.getStatusAsString().equals("NEGATIVE");
+    private static final Predicate<Person> BY_HRW = person -> person.getStatusAsString().equals("HRW");
+    private static final Predicate<Person> BY_HRN = person -> person.getStatusAsString().equals("HRN");
+
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        List<Person> lastShownList = model.getFilteredPersonList();
+        StringBuilder answer = new StringBuilder(MESSAGE_SUMMARISE_PERSON_SUCCESS);
+
+        for (String facultyName : FACULTIES) {
```

Comment:
Just a suggestion, it may be good to abstract out this portion of the method to prevent the method from being too long 😄

Reply:
Noted! I will make amendments to apply better abstractions here.